sql/core/src/test/scala/org/apache/spark/sql/TPCDSCollationQueryTestSuite.scala

```diff
@@ -0,0 +1,214 @@
+      spark.sql(s"CREATE DATABASE `${check.dbName}`")
+      withDB(check.dbName) {
+        tableNames.foreach(tableName => {
+          val columns = tableColumns(tableName)
+            .split("\n")
+            .filter(_.trim.nonEmpty)
+            .map { column =>
+              if (column.trim.split("\\s+").length != 2) {
+                throw new IllegalArgumentException(s"Invalid column definition: $column")
+              }
+              val Array(name, colType) = column.trim.split("\\s+")
+              (name, colType.replaceAll(",$", ""))
+            }
+
+          spark.sql(
+            s"""
+               |CREATE TABLE `$tableName` (${collateStringColumns(columns, check.collation)})
+               |USING parquet
+               |""".stripMargin)
+
+          val transformedColumns = columns.map { case (name, colType) =>
+            if (isTextColumn(colType)) {
+              s"${check.columnTransform}($name) AS $name"
+            } else {
+              name
+            }
+          }.mkString(", ")
+
+          spark.sql(
+            s"""
+               |INSERT INTO TABLE `$tableName`
+               |SELECT $transformedColumns
+               |FROM parquet.`${tpcdsDataPath.get}/$tableName`
+               |""".stripMargin)
+        })
+      }
+    })
+  }
+
+  override def dropTables(): Unit =
+    checks.flatten.foreach(check => {
+      withDB(check.dbName)(super.dropTables())
+      spark.sql(s"DROP DATABASE `${check.dbName}`")
+    })
+
+  private def collateStringColumns(
+      columns: Array[(String, String)],
+      collation: String): String = {
+    columns
+      .map { case (name, colType) =>
+        if (isTextColumn(colType)) {
+          s"$name STRING COLLATE $collation"
+        } else {
+          s"$name $colType"
+        }
+      }
+      .mkString(",\n")
+  }
+
+  private def isTextColumn(columnType: String): Boolean = {
+    columnType.toUpperCase(Locale.ROOT).contains("CHAR")
+  }
+
+  private def runQuery(query: String, conf: Map[String, String]): Unit = {
+    withSQLConf(conf.toSeq: _*) {
+      try {
+        checks.foreach(batch => {
+          val res = batch.map(check =>
+            withDB(check.dbName)(getQueryOutput(check.queryTransform(query)).toLowerCase()))
+          if (res.nonEmpty) res.foreach(currRes => assertResult(currRes)(res.head))
+        })
+      } catch {
+        case e: Throwable =>
+          val configs = conf.map { case (k, v) =>
+            s"$k=$v"
+          }
+          throw new Exception(s"${e.getMessage}\nError using configs:\n${configs.mkString("\n")}")
+      }
+    }
+  }
+
+  private def getQueryOutput(query: String): String = {
+    val (_, output) = handleExceptions(getNormalizedQueryExecutionResult(spark, query))
+    output.mkString("\n").replaceAll("\\s+$", "")
+  }
+
+  // skip q91 due to use of like expression which is not supported with collations yet
+  override def excludedTpcdsQueries: Set[String] = super.excludedTpcdsQueries ++ Set("q91")
```

Comment:
what do you think about writing the output to golden files like the regular tpcds test does?

that way we won't have to skip this query

Reply:
The idea to omit golden files was that the output is for the most part just lowercase version of regular TPC-DS outputs which are already checked-in and would be kinda duplicate.

Not sure though how adding golden files would help for this particular case as they first need to be generated but this query errors out because it contains like expression which still doesn't support collations.